Bug: Property 'aborted' does not exist on type 'AbortSignal'

### What version of Bun is running?

1.2.21+7c45ed97d

### What platform is your computer?

Linux 6.16.4-arch1-1 x86_64 unknown

### What steps can reproduce the bug?

```bash
# Clone the repro repo
git clone https://github.com/nikelborm/bun-signal-aborted-repro.git

cd bun-signal-aborted-repro

# Install dependencies
bun install

# Works fine as expected
bun run index.ts
# Outputs:
# 1 signal.aborted:  false
# 2 signal.aborted:  true

# Fails
./node_modules/.bin/tsc
# Outputs
# index.ts:3:42 - error TS2339: Property 'aborted' does not exist on type 'AbortSignal'.
#
# 3 console.log('1 signal.aborted: ', signal.aborted);
#                                            ~~~~~~~
#
# index.ts:6:44 - error TS2339: Property 'aborted' does not exist on type 'AbortSignal'.
#
# 6   console.log('2 signal.aborted: ', signal.aborted);
#                                              ~~~~~~~
#
#
# Found 2 errors in the same file, starting at: index.ts:3
```


### What is the expected behavior?

The code below should not produce any ts errors

```ts
const signal = AbortSignal.timeout(20);

console.log('1 signal.aborted: ', signal.aborted);

setTimeout(() => {
  console.log('2 signal.aborted: ', signal.aborted);
}, 40);
```



### What do you see instead?

It does: `Property 'aborted' does not exist on type 'AbortSignal'.`

### Additional information

Repro: https://github.com/nikelborm/bun-signal-aborted-repro.git

--- a/packages/bun-types/globals.d.ts
+++ b/packages/bun-types/globals.d.ts
@@ -1564,6 +1564,12 @@ declare var AbortController: Bun.__internal.UseLibDomIfAvailable<
   }
 >;
 
+interface AbortSignal extends EventTarget {
+  readonly aborted: boolean;
+  onabort: ((this: AbortSignal, ev: Event) => any) | null;
+  readonly reason: any;
+  throwIfAborted(): void;
+}
 declare var AbortSignal: Bun.__internal.UseLibDomIfAvailable<
   "AbortSignal",
   {
@@ -1948,3 +1954,21 @@ declare namespace fetch {
   ): void;
 }
 //#endregion
+
+interface RegExpConstructor {
+  /**
+   * Escapes any potential regex syntax characters in a string, and returns a
+   * new string that can be safely used as a literal pattern for the RegExp()
+   * constructor.
+   *
+   * [MDN Reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/escape)
+   *
+   * @example
+   * ```ts
+   * const re = new RegExp(RegExp.escape("foo.bar"));
+   * re.test("foo.bar"); // true
+   * re.test("foo!bar"); // false
+   * ```
+   */
+  escape(string: string): string;
+}

--- a/test/integration/bun-types/fixture/index.ts
+++ b/test/integration/bun-types/fixture/index.ts
@@ -455,3 +455,9 @@ Bun.serve({
     cert,
   },
 });
+
+const signal = AbortSignal.timeout(1000);
+expectType(signal).is<AbortSignal>();
+expectType(signal.aborted).is<boolean>();
+
+expectType(RegExp.escape("foo.bar")).is<string>();

